mobile/library/cc/engine_builder.cc

```diff
@@ -222,43 +222,50 @@ EngineBuilder& EngineBuilder::enforceTrustChainVerification(bool trust_chain_ver
   return *this;
 }
 
-EngineBuilder& EngineBuilder::setNodeId(std::string node_id) {
+EngineBuilder& EngineBuilder::setNodeId(std::string node_id, bool use_node_id) {
   node_id_ = std::move(node_id);
+  use_node_id_ = use_node_id;
   return *this;
 }
 
-EngineBuilder& EngineBuilder::setNodeLocality(const NodeLocality& node_locality) {
+EngineBuilder& EngineBuilder::setNodeLocality(const NodeLocality& node_locality,
+                                              bool use_node_locality) {
   node_locality_ = node_locality;
+  use_node_locality_ = use_node_locality;
   return *this;
 }
 
-EngineBuilder& EngineBuilder::setAggregatedDiscoveryService(const std::string& address,
-                                                            const int port, std::string jwt_token,
+EngineBuilder& EngineBuilder::setAggregatedDiscoveryService(std::string address, const int port,
+                                                            std::string jwt_token,
                                                             const int jwt_token_lifetime_seconds,
-                                                            std::string ssl_root_certs) {
+                                                            std::string ssl_root_certs,
+                                                            bool use_ads) {
 #ifndef ENVOY_GOOGLE_GRPC
   throw std::runtime_error("google_grpc must be enabled in bazel to use ADS");
 #endif
   ads_address_ = address;
   ads_port_ = port;
   ads_jwt_token_ = std::move(jwt_token);
-  ads_jwt_token_lifetime_seconds_ = jwt_token_lifetime_seconds;
+  ads_jwt_token_lifetime_seconds_ =
+      jwt_token_lifetime_seconds == 0 ? DefaultJwtTokenLifetimeSeconds : jwt_token_lifetime_seconds;
```

Comment:
Moved the timeout defaults into the function bodies because we're passing in 0 from other languages to represent null (no easy way to pass nulls around).